addons/pos_restaurant/tests/test_frontend.py

```diff
@@ -37,6 +39,19 @@ def setUpClass(cls):
             'type': 'cash',
             'company_id': main_company.id,
         })
+        cls.payment_provider = cls.provider # The dummy_provider used by the tests of the 'payment' module.
+
+        cls.payment_provider_old_company_id = cls.payment_provider.company_id.id
+        cls.payment_provider_old_journal_id = cls.payment_provider.journal_id.id
+        cls.payment_provider.write({
+            'company_id': cls.company.id,
+            'journal_id': cls.company_data['default_journal_bank'].id,
+        })
+        cls.online_payment_method = cls.env['pos.payment.method'].create({
+            'name': 'Online payment',
+            'is_online_payment': True,
+            'online_payment_provider_ids': [Command.set([cls.payment_provider.id])],
+        })
```

Comment:
Hello @aamo-odoo thanks for the changes, we're almost there. 

We still have an issue with this when we only have `pos_restaurant` installed. Because in this case `'pos.payment.methd' doesn't have the field `is_online_payment`. This isn't caught by the runbot right now but it will fail the nighlty build when they try each modules by themselves.

I had a db with only `pos_restaurant` and an error popped up when launching `test_01_pos_restaurant` for example.  

It's best if you make the setup in `addons/pos_online_payment_self_order/tests/test_self_order_frontend.py` and change nothing here if possible.